pkg/operator/hive/hive.go

```diff
@@ -464,6 +492,76 @@ func (r *ReconcileHiveConfig) includeGlobalPullSecret(hLog log.FieldLogger, inst
 	hiveContainer.Env = append(hiveContainer.Env, globalPullSecretEnvVar)
 }
 
+func (r *ReconcileHiveConfig) includeHivePrivateImagePullSecret(hLog log.FieldLogger, instance *hivev1.HiveConfig, hiveContainer *corev1.Container) {
+	if !hiveConfigHasValidImagePullSecretReference(instance) {
+		hLog.Debug("HiveImagePullSecret is not provided in HiveConfig, it will not be deployed")
+		return
+	}
+
+	hiveImagePullSecretEnvVar := corev1.EnvVar{
+		Name:  constants.HivePrivateImagePullSecret,
+		Value: instance.Spec.HiveImagePullSecretRef.Name,
+	}
+	hiveContainer.Env = append(hiveContainer.Env, hiveImagePullSecretEnvVar)
+}
+
+func (r *ReconcileHiveConfig) dynamicCopyHiveImagePullSecret(hLog log.FieldLogger, instance *hivev1.HiveConfig) error {
```

Comment:
Okay, yeah, let's clean this up.

For the getting, you should be able to use the existing controller-runtime-esque wrapper [here](https://github.com/openshift/hive/blob/b69fcfef87fc9f72cb021b4bfa3f86e59a46e2ef/pkg/operator/hive/dynamicclient.go#L66). Example [Get()](https://github.com/openshift/hive/blob/b69fcfef87fc9f72cb021b4bfa3f86e59a46e2ef/pkg/operator/hive/hive.go#L589).

For the creating, you should be able to write a similar `Create()` wrapper in that file.

The end result should be that you can make this func feel a lot more like it would if `r` had/was a real c-r client.